cluster/service.go

```diff
@@ -147,6 +147,7 @@ func (s *clusterService) refreshCache(ctx context.Context) error {
 			return errors.Wrapf(err, "Unable to resolve token for cluster %v", cluster.APIURL)
 		}
 		// verify the token
+		s.clientOptions = append(s.clientOptions, configuration.WithInsecureSkipTLSVerify())
```

Comment:
Looks like a very odd location to append this. Move it to the creator of this instead. Or some 'default options list' in constructor or similar.